tests/Nox.Types.Tests/Types/CurrencyCode3/CurrencyCode3Tests.cs

```diff
@@ -0,0 +1,61 @@
+﻿using FluentAssertions;
+
+namespace Nox.Types.Tests.Types.CountryCode3;
+public class CurrencyCode3Tests
+{
+    [Theory]
+    [ClassData(typeof(CurrencyCodeData))]
+    public void CurrencyCode3_Constructor_ReturnsSameValue_AllCurrencies(string currencyCode3String)
+    {
+        var currencyCode3 = CurrencyCode3.From(currencyCode3String);
+
+        currencyCode3.Value.Should().Be(currencyCode3String);
+    }
+
+    [Fact]
+    public void CurrencyCode3_Constructor_WithUnsupportedCurrencyCode3_ThrowsValidationException()
+    {
+        var action = () => CurrencyCode3.From("ABC");
+
+        action.Should().Throw<TypeValidationException>()
+            .And.Errors.Should().BeEquivalentTo(new[] { new ValidationFailure("Value", "Could not create a Nox CurrencyCode3 type with unsupported value 'ABC'.") });
+    }
+
+    [Fact]
+    public void CurrencyCode3_Equality_Tests()
+    {
+        var currencyCode3_1 = CurrencyCode3.From("USD");
+
+        var currencyCode3_2 = CurrencyCode3.From("USD");
+
+        currencyCode3_1.Should().Be(currencyCode3_2);
+    }
+
+    [Fact]
+    public void CurrencyCode3_CaseInsensitiveEquality_Tests()
```

Comment:
this test is failing in the CI, but should be valid / its correctly implemented.

I think we should do a ToUpper in From  method because we assume that Currency is always in upper case, do y agree?

Reply:
I totally agree. Updating them